src/AppInstallerCommonCore/Manifest/YamlWriter.cpp

```diff
@@ -0,0 +1,661 @@
+            WRITE_BOOL_PROPERTY(out, DisplayInstallWarnings, installer.DisplayInstallWarnings);
+            WRITE_PROPERTY_IF_EXISTS(out, MinimumOSVersion, installer.MinOSVersion);
+            WRITE_PROPERTY_IF_EXISTS(out, ProductCode, installer.ProductCode);
+            WRITE_PROPERTY_IF_EXISTS(out, UpgradeBehavior, UpdateBehaviorToString(installer.UpdateBehavior));
+
+            ProcessSequence(out, Capabilities, installer.Capabilities);
+            ProcessSequence(out, Commands, installer.Commands);
+            ProcessSequence(out, FileExtensions, installer.FileExtensions);
+            ProcessSequence(out, Protocols, installer.Protocols);
+            ProcessSequence(out, RestrictedCapabilities, installer.RestrictedCapabilities);
+
+            ProcessAppsAndFeaturesEntries(out, installer.AppsAndFeaturesEntries);
+            ProcessDependencies(out, installer.Dependencies);
+            ProcessExpectedReturnCodes(out, installer.ExpectedReturnCodes);
+            ProcessInstallerSwitches(out, installer.Switches);
+            ProcessInstallationMetadata(out, installer.InstallationMetadata);
+            ProcessInstallerSuccessCodes(out, installer.InstallerSuccessCodes);
+            ProcessInstallModes(out, installer.InstallModes);
+            ProcessMarkets(out, installer.Markets);
+            ProcessNestedInstallerFiles(out, installer.NestedInstallerFiles);
+            ProcessPlatforms(out, installer.Platform);
+            ProcessUnsupportedArguments(out, installer.UnsupportedArguments);
+            ProcessUnsupportedOSArchitecture(out, installer.UnsupportedOSArchitectures);
+        }
+
+        void ProcessInstaller(YAML::Emitter& out, const ManifestInstaller& installer)
+        {
+            out << YAML::Key << Installers;
+            out << YAML::BeginSeq;
+            out << YAML::BeginMap;
+            ProcessInstallerFields(out, installer);
+            out << YAML::EndMap;
+            out << YAML::EndSeq;
+        }
+    }
+
+#ifndef AICLI_DISABLE_TEST_HOOKS
+    static ManifestTypeEnum* s_ExportedManifestTypeResult_TestHook_Override = nullptr;
+
+    void TestHook_SetExportedManifestTypeResult_Override(ManifestTypeEnum* manifestType)
+    {
+        s_ExportedManifestTypeResult_TestHook_Override = manifestType;
+    }
+#endif
+
+    std::string YamlWriter::ManifestToYamlString(const Manifest& manifest, const ManifestInstaller& installer)
+    {
+        YAML::Emitter out;
+        out << YAML::BeginMap;
+        WRITE_PROPERTY_IF_EXISTS(out, PackageIdentifier, manifest.Id);
+        WRITE_PROPERTY_IF_EXISTS(out, PackageVersion, manifest.Version);
+        WRITE_PROPERTY_IF_EXISTS(out, Channel, manifest.Channel);
+        WRITE_PROPERTY_IF_EXISTS(out, Moniker, manifest.Moniker);
+        ProcessLocaleFields(out, manifest.DefaultLocalization);
+
+        // Handles default installer fields in the root node.
+        ProcessInstallerFields(out, manifest.DefaultInstallerInfo);
```

Comment:
DefaultInstaller is not needed, default info is already populated to individual installer if applicable.

Reply:
I left this as is. Some fields like the installer's package locale don't get populated in the root of the manifest so the unit test will fail.

Reply:
I think we should update the test to ignore DefaultInstallerInfo check for the exported manifest case.

Reply:
Changed to ignoring default installer info as discussed.